lib/worker.js

```diff
@@ -330,12 +344,39 @@ class AssetComputeWorker {
         }
     }
 
+    sendEventsBeforeActionTimeout() {
+        return setTimeout(
+            async () => {
+                console.log(`Action is about to timeout in ${timeUntilTimeout()}ms Sending IO events before timeout`);
+                // ensure failure events are sent for any non successful rendition before timeout
+                if(this.renditions){
+                    for (const rendition of this.renditions) {
+                        if (!rendition.eventSent) {
+                            await this.events.sendEvent(EVENT_RENDITION_FAILED, {
+                                rendition: rendition.instructionsForEvent(),
+                                errorReason: Reason.GenericError,
+                                errorMessage: "Timeout",
+                                activationIds: this.params.customWorker ? [process.env.__OW_ACTIVATION_ID] : undefined
+                            });
```

Comment:
We also need to send the corresponding rendition failed metrics (`handleError()`). See https://github.com/adobe/asset-compute-sdk/blob/master/lib/worker.js#L317-L329

(This somewhat makes the `timeout` metric duplicated. But it doesn't hurt to have the two. We can always disable later once we are in production and actually see the data in NR and think it's superfluous).

Reply:
Done, see below. Since we will not really know what part of the processing the code is in when it times out, it is hard to determine which metrics to add (processingDuration etc.) so all I added was `duration` for now